tests/Aspire.Hosting.Garnet.Tests/GarnetFunctionalTests.cs

```diff
@@ -59,4 +55,158 @@ await pipeline.ExecuteAsync(async token =>
+                            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(10) })
+                            .Build();
+        string? volumeName = null;
+        string? bindMountPath = null;
+
+        try
+        {
+            var builder1 = TestDistributedApplicationBuilder.Create().WithTestAndResourceLogging(testOutputHelper);
+            var garnet1 = builder1.AddGarnet("garnet");
+
+            if (useVolume)
+            {
+                // Use a deterministic volume name to prevent them from exhausting the machines if deletion fails
+                volumeName = VolumeNameGenerator.CreateVolumeName(garnet1, nameof(WithDataShouldPersistStateBetweenUsages));
+
+                // if the volume already exists (because of a crashing previous run), try to delete it
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+                garnet1.WithDataVolume(volumeName);
+            }
+            else
+            {
+                bindMountPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+
+                if (!Directory.Exists(bindMountPath))
+                {
+                    if (OperatingSystem.IsWindows())
+                    {
+                        Directory.CreateDirectory(bindMountPath);
+                    }
+                    else
+                    {
+                        Directory.CreateDirectory(bindMountPath, UnixFileMode.UserRead | UnixFileMode.UserWrite | UnixFileMode.GroupRead | UnixFileMode.OtherRead);
+                    }
+                }
+                garnet1.WithDataBindMount(bindMountPath);
+            }
+
+            using (var app = builder1.Build())
+            {
+                await app.StartAsync();
+                try
+                {
+                    var hb = Host.CreateApplicationBuilder();
+
+                    hb.Configuration[$"ConnectionStrings:{garnet1.Resource.Name}"] = $"{await garnet1.Resource.ConnectionStringExpression.GetValueAsync(default)},allowAdmin = true";
+
+                    hb.AddRedisClient("garnet");
+
+                    using (var host = hb.Build())
+                    {
+                        await host.StartAsync();
+
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            var redisClient = host.Services.GetRequiredService<IConnectionMultiplexer>();
+
+                            var db = redisClient.GetDatabase();
+
+                            await db.StringSetAsync("key", "value");
+                            var value = await db.StringGetAsync("key");
+
+                            Assert.Equal("value", value);
+                            // Force Redis to save the keys (snapshotting)
+                            // c.f. https://redis.io/docs/latest/operate/oss_and_stack/management/persistence/
+
+#pragma warning disable CS0618 // Type or member is obsolete
```

Comment:
What member is obsolete?

Reply:
Because of blocking. Redis client suggested to use BackgroundSave but for garnet it take too long  that i expect. So i use ForgroundSave to insure data persiste.

Reply:
Can you add a comment?

Is there some way to wait for the save to occur? For example using:

https://github.com/dotnet/aspire/blob/ccec365c9aead12e2ab4f833e02e1b2844312e9a/tests/Aspire.Hosting.Tests/Utils/LoggerNotificationExtensions.cs#L31-L39

Reply:
I'm not sure garnet log something useful around this but i will try.

Reply:
I used following the code to force Garnet to save keys.

```csharp
      // Force Garnet to save the keys
      // c.f. https://microsoft.github.io/garnet/docs/commands/checkpoint#save
      await db.ExecuteAsync("SAVE");
```

I could not find any text to wait for the save to occur.